### PR TITLE
Optionally clear CMOS before exposure

### PR DESCRIPTION
**Pull request by const86**
CMOS is normally reset just before every exposure. But sometimes residual charge may remain
and pollute new image. Running dummy short exposures just before starting a real one is found
to help in some situations. For example, it reduces ghosts of stars after dithering.
Disabled by default, user may choose to blink any number of times.

Validated on my ASI294MC Pro. Each blink takes about 300-400ms of extra time.

See extensive discussion here: https://www.cloudynights.com/topic/662887-not-rbi-on-cmos-but-this-looks-similar/

### CODE REVIEW

File: indi-asi/asi_ccd.cpp [hunk 1]
```diff
@@ -1023,6 +1036,51 @@ bool ASICCD::StartExposure(float duration)
 {
     ASI_ERROR_CODE errCode = ASI_SUCCESS;
 
+    long blinks = BlinkN[0].value;
+    if (blinks > 0)
+    {
+        LOGF_INFO("Blinking %ld time(s) before exposure", blinks);
+
+        const long duration = 0;
+        errCode = ASISetControlValue(m_camInfo->CameraID, ASI_EXPOSURE, duration, ASI_FALSE);
```
Comment (knro) [inline indi-asi/asi_ccd.cpp:1045]: This is using duration which is the exposure time? isn't this suppose to be short 100ms or something?

Comment (const86) [reply indi-asi/asi_ccd.cpp:1045]: Yes, it is exposure time. 0 seems to work for me. I have zero experience with ASI and CMOS idiosyncrasies... so I can't really prove it is the best one. Probably, I can make it configurable too.

Comment (const86) [reply indi-asi/asi_ccd.cpp:1046]: ...done.

**Comment by const86** [comment]
> Thank you! Need to add BlinkNP to saveConfig function as well so that it is remembered.

Done.

File: indi-asi/asi_ccd.cpp [hunk 2]
```diff
@@ -2219,5 +2220,8 @@ bool ASICCD::saveConfigItems(FILE *fp)
     if (VideoFormatSP.nsp > 0)
         IUSaveConfigSwitch(fp, &VideoFormatSP);
 
+    if (BlinkNP.nnp > 0)
```
Comment (knro) [inline indi-asi/asi_ccd.cpp:2223]: That's always going to be 1. You probably meant
if (BlinkN[0].value > 0)

File: indi-asi/asi_ccd.h [hunk 3]
```diff
@@ -154,7 +154,7 @@ class ASICCD : public INDI::CCD
         uint8_t rememberVideoFormat = { 0 };
         ASI_IMG_TYPE currentVideoFormat;
 
-        INumber BlinkN[1];
+        INumber BlinkN[2];
```
Comment (knro) [inline indi-asi/asi_ccd.h:157]: It's recommended to add enums to make this clearer.

enum
{
  BLINK_COUNT,
  BLINK_DURATION
};

then in code when you use BlinkN[BLINK_COUNT].value for example, it's clear to which element we are referring to.

Comment (const86) [reply indi-asi/asi_ccd.h:162]: Done

**Comment by const86** [comment]
> This change also required a version change in CMakelists.txt and another in debian/indi-asi/changelog to mark the changes.

Done.

**Pull request merged**